sympy/stats/rv.py

```diff
@@ -1514,3 +1531,40 @@ def _symbol_converter(sym):
     if not isinstance(sym, Symbol):
         raise TypeError("%s is neither a Symbol nor a string"%(sym))
     return sym
+
+
+@singledispatch
```

Comment:
The question is whether the master definition has to be in this file. I would suggest a main file where is_random is defined, then overload the various classes in the other files.

Reply:
The advantage of singledispatch is to avoid writing code in sympy/core which is needed in other modules. The core of SymPy has been hacked too much because of requirements of external modules.

Single dispatch still works as a class method in a certain way, so you can spread the definitions over many files.

Reply:
>  would suggest a main file where is_random is defined

Actually, IMO `rv.py` can be considered as the core of stats as it contains all the basic classes and functions of stats. So, if we want to split it over files, we can split it, but I suggest keeping main definition and few basic registers of classes like `Expr` in `rv.py`.